### ISSUE TITLE
Cannot get User context

### ISSUE THREAD

**Issue by gmgarrison** [getsentry/sentry-laravel#949]
### How do you use Sentry?

Sentry SaaS (sentry.io)

### SDK version

4.9

### Laravel version

10.48.22

### Steps to reproduce

I found the directions on this page: https://docs.sentry.io/platforms/php/guides/laravel/enriching-events/identify-user/

1. I created a `SentryUserContext` middleware and placed the `\configureScope` code in the `handle()` method.
2. I added the middleware to the HTTP kernel.
3. I ensured that `send_default_pii` is set to `true` in my `sentry.php` config file

**Middleware code:**

```php
class SentryUserScope
{
    public function handle(Request $request, Closure $next)
    {
        if ($user = $request->user()) {
            \Sentry\configureScope(function (\Sentry\State\Scope $scope) use ($user): void {
                $scope->setUser([
                    'email' => $user->email,
                    'id' => $user->id,
                ]);
            });
        }

        return $next($request);
    }
}
```

### Expected result

I expected something more than just the IP address to be in the "user" side panel on an Issue.

I'm also not sure how to debug it to see where it might be failing?

### Actual result

![Image](https://github.com/user-attachments/assets/e7271e0d-9182-4962-af6d-20b97608f675)

**Comment by stayallive** [comment]
If you just care about the e-mail and user ID you do not need a custom middleware and setting `send_default_pii` is enough: https://docs.sentry.io/platforms/php/guides/laravel/data-management/data-collected/#information-about-logged-in-user. If you have added this is a global middleware it probably doesn't work because the session has not been started yet. 

This is information is only collected after the user is actually authenticated. So after Laravel has started the session etc. Is it possible you are looking at an error that is occurring before authentication has been loaded by the framework?

**Comment by gmgarrison** [comment]
Thank you for the reply.

Unfortunately, no, I don't think it's possible that the error would be before session authentication. I've gotten errors from my own Laravel Controllers that are actively using Auth::user() but I never get any user context in Sentry.

To give slightly more context, I've had Sentry deployed for months and with an architecture change, it's now more important to see user context in our Sentry errors so now I'm finally trying to chase down what's going on here.

**Comment by stayallive** [comment]
Is there anything in your application you can think of that causes `Illuminate\Auth\Events\Authenticated` not to fire? Are you using a special kind of guard or maybe even a custom guard?

**Comment by gmgarrison** [comment]
Just the standard web guard with file-based cookies. I admit I don’t have too much experience with many Laravel internals though. I had to google what a guard is. 

Can you think of a snippet to run in my code to try to figure out what’s (not?) happening?

**Comment by stayallive** [comment]
In the `SentryUserScope` you could try to `dd($user)` to see if the code even ends up there.

This assumes you are testing locally of course, do not do that in production 😉

**Comment by gmgarrison** [comment]
Well, I'm not sure what's happened but within the last 24 hours I now have user ID's showing up in Sentry. I've made lots of code changes, but not related to this issue (or at least not that I realized!). I have my SentryUserScope middleware still in place but all of the code is commented out.

I wish I understood a little better what was going on, but for the moment I'll take the win. Thank you for your help along the way!
